runtime/libffi/z/sysvz64.s

```diff
@@ -527,9 +884,141 @@ RLL      DS 0H
          B RET
 
 RV       DS 0H
-RS       DS 0H
          B RET
 
+RS       DS 0H
+         LG 7,(2176+(((DSASZ+31)/32)*32)+24)(,4)
+         LG 14,(2176+(((DSASZ+31)/32)*32)+8)(,4)
+         LG 14,0(,14)           ecif->cif
+         LG 9,16(,14)           cif->rtype
+         LG 14,0(,9)            rtype->size
+         CGIBH 14,24,RSP
+
+* now we need to determine how to return the struct
+* this label determines if the struct is <= 24 bytes
+* if it is we return using gprs/frprs
+* we then loop over rtype->elements
+* to determine whether this struct is all floating point
+* or if it is contains integral types
+* if we have just floating point types use FPRs as appropriate
+* but only if we have just the same type of float
+* a struct mixing float, double and long double will use GPRs
+* i have no idea why it is done this way, but it is
```

Comment:
Comment can be cleaned up to briefly mention the rules for return values ?
i.e, 
- Integral <= 64 bytes are widened to 64 bits and returned in R3
- Complex type (Two floats in struct / Two double in Structs) or floating point type should use FPR0/2/4/6
- <= 24 bytes struct in R1/R2/R3 
- >24 in the memory  pointed by the hidden parameter